operators/src/adapters/raster_overlap_adapter.rs

```diff
@@ -0,0 +1,528 @@
+where
+    T: Pixel,
+    Q: RasterQueryProcessor<RasterType = T>,
+    QC: QueryContext,
+    XF: Clone + Fn(TileInformation, QueryRectangle) -> Result<(RasterTile2D<T>, X)>,
+    X: Clone + Send,
+
+    FoldFut: Future<Output = Result<(RasterTile2D<T>, X)>>,
+    FoldF: Clone + Fn(Result<(RasterTile2D<T>, X)>, Result<RasterTile2D<T>>) -> FoldFut,
+
+    QF: Clone + Fn(&TileInformation, &QueryRectangle, TimeInstance) -> Result<QueryRectangle>,
+{
+    type Item = Result<RasterTile2D<T>>;
+
+    fn poll_next(
+        self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        if self.is_terminated() {
+            return Poll::Ready(None);
+        }
+
+        let mut this = self.project();
+
+        if this.running_future.as_ref().is_none() {
+            if *this.current_spatial_tile >= this.spatial_tiles.len() {
+                *this.current_spatial_tile = 0;
+                *this.time_start = *this.time_end;
+                *this.time_end = None;
+            }
+
+            if let Some(t_start) = *this.time_start {
+                if t_start >= this.query_rect.time_interval.end() {
+                    *this.ended = true;
+                } else {
+                    // TODO: maybe this whole block should also be a future!
+                    let current_spatial_tile_info = this.spatial_tiles[*this.current_spatial_tile];
+
+                    let fold_tile_spec = TileInformation {
+                        tile_size_in_pixels: current_spatial_tile_info.tile_size_in_pixels,
+                        global_tile_position: current_spatial_tile_info.global_tile_position,
+                        global_geo_transform: current_spatial_tile_info.global_geo_transform,
+                    };
+
+                    let tqr = (this.tile_query_rewrite)(
+                        &current_spatial_tile_info,
+                        &this.query_rect,
+                        t_start,
+                    )?;
+
+                    let qs = this.source.raster_query(tqr, *this.query_ctx)?;
+
+                    let accu = (this.accu_creator)(fold_tile_spec, tqr);
+
+                    let ttf = qs.fold(accu, this.fold_fn.clone());
+
+                    this.running_future.set(Some(ttf));
+                }
+            }
+        }
+
+        let rv = match this.running_future.as_mut().as_pin_mut() {
+            Some(fut) => ready!(fut.poll(cx)),
+            None => return Poll::Ready(None),
+        };
+
+        let r = match rv {
+            Ok(tile) => tile,
+            Err(err) => return Poll::Ready(Some(Err(err))),
+        };
+
+        // set the running future to None --> will create a new one in the next call
+        this.running_future.set(None);
+
+        // update the end_time from the produced tile (should/must not change within a tile run?)
+        let t_end = r.0.time.end();
+        let _old_t_end = this.time_end.replace(t_end);
```

Comment:
This is unused?

Reply:
yes thats why it has a _ prefix

Reply:
Okay, but why do you replace something and don't use it?

Reply:
i need to set this.time_end

Reply:
Ah okay, you could just say
```suggestion
        this.time_end.replace(t_end);
```
, right?

Reply:
8cd300c